Site1.Master.cs

```diff
@@ -11,7 +11,45 @@ public partial class Site1 : System.Web.UI.MasterPage
     {
         protected void Page_Load(object sender, EventArgs e)
         {
+            try
+            {
+                if (Session["role"].Equals(""))
+                {
+                    LinkButton1.Visible = true; //user login link button
+                    LinkButton2.Visible = true; //sing up link button
 
+                    LinkButton3.Visible = false; //logout link button
+                    LinkButton7.Visible = false; //hello user login link button
+
+                    LinkButton6.Visible = true; //admin login link button
+                }
+                else if (Session["role"].Equals("user"))
+                {
+                    LinkButton1.Visible = false; //user login link button
+                    LinkButton2.Visible = false; //sing up link button
+
+                    LinkButton3.Visible = true; //logout link button
+                    LinkButton7.Visible = true; //hello user login link button
+                    LinkButton7.Text = "Hello " + Session["username"].ToString();
+
+                    LinkButton6.Visible = true; //admin login link button
+                }
+                else if (Session["role"].Equals("admin"))
+                {
+                    LinkButton1.Visible = false; //user login link button
+                    LinkButton2.Visible = false; //sing up link button
+
+                    LinkButton3.Visible = true; //logout link button
+                    LinkButton7.Visible = true; //hello user login link button
+                    LinkButton7.Text = "Hello Admin";
+
+                    LinkButton6.Visible = false; //admin login link button
+                }
+            }
+            catch(Exception ex)
```

Comment:
### Nothing being done to resolve the problem. 
When something goes wrong within the _try_ statement, it will come here. 

Right now your catch is empty, so you need to decide what you want to do with the exceptions that happen. 

Resolve, skip, rollback, undo, provide alternative solutions. All of these are viable options.